typescript/cli/src/tests/commands/core.ts

```diff
@@ -30,6 +30,7 @@ export async function hyperlaneCoreRead(chain: string, coreOutputPath: string) {
         --registry ${REGISTRY_PATH} \
         --config ${coreOutputPath} \
         --chain ${chain} \
+        --key ${ANVIL_KEY} \
```

Comment:
For chains such as ZKSync, its necessary to have Signer to read state from the chain. For Ethereum chains, we can create random key for Signer, so the users don't need to pass --key, but not sure if this can work for StarkNet

Reply:
Instead of making read commands require a signer, can you instead prompt for signer info at the last step when the read happens to be to a ZKSync-like chain? If that would be difficult, then let's at least make sure this arg is still optional

Reply:
Hmm lets then not think about zksync for now, and make a separate PR just for zksync. We can always use `Wallet.createRandom()` from 'zksync-ethers' for reading